src/Tasks/ResolveManifestFiles.cs

```diff
@@ -380,6 +382,29 @@ private void GetOutputAssemblies(List<PublishInfo> publishInfos, List<ITaskItem>
                 {
                     if (!IsFiltered(item))
                     {
+                        // ClickOnce for .NET 4.X should not publish duplicate satellite assemblies.
+                        // This will cause ClickOnce install to fail. This can happen if some package
+                        // decides to publish the en-us resource assemblies for other locales also.
+                        if (!LauncherBasedDeployment && satelliteRefAssemblyMap.ContainsItem(item))
+                        {
+                            Debug.Assert(false, $"Duplicate satellite assembly '{item.ItemSpec}' skipped in _managedAssemblies");
+                            continue;
+                        }
+
+                        // If we get a resource assembly in managed references, determine whether to be publish it based on _targetCulture
+                        AssemblyIdentity identity = AssemblyIdentity.FromManagedAssembly(item.ItemSpec);
+                        if (identity != null && !String.Equals(identity.Culture, "neutral", StringComparison.Ordinal))
+                        {
+                            CultureInfo satelliteCulture = GetItemCulture(item);
+                            if (!PublishFlags.IsSatelliteIncludedByDefault(satelliteCulture, _targetCulture, _includeAllSatellites))
+                            {
+                                continue;
+                            }
+                            else
+                            {
+                                satelliteRefAssemblyMap.Add(item);
+                            }
```

Comment:
for readability I'd prefer

```suggestion
                            if (PublishFlags.IsSatelliteIncludedByDefault(satelliteCulture, _targetCulture, _includeAllSatellites))
                            {
                                satelliteRefAssemblyMap.Add(item);
                            }
                            else
                            {
                                continue;
                            }
```

Reply:
fixed.